articles/configure-your-commit-messages.mdx

```diff
@@ -8,162 +8,214 @@ published: true
+  - template_string___2___: "{message}"
+```
+<comp.Annotation number="1">
+  The `commit_templates` key contains all the commit message templates in an array.
+</comp.Annotation>
+<comp.Annotation number="2">
+  The `template_string` key contains the template for your commit. In this example the template string uses the dynamic placeholder `{message}`. As the key "message" is undefined, CloudCannon will provide a text input.
+</comp.Annotation>
+</comp.MultiCodeBlock>
 
-**Commit placeholders** insert data related to the commit, and are always surrounded by `[ ]` brackets. The available commit placeholders are:
+Once you have saved the changes to your global configuration file, the *Review changes* modal in CloudCannon will update to include a commit message.
 
-* `changes` is a bulleted list of all changes (edits, additions and deletions)
-* `date` is the current date
-* `author` is the email address of the person making the commit
-* `default_commit_message` is the default commit message that CloudCannon would generate. This will resolve to a sentence like "Added/Updated X file(s)".
+<comp.DocsImage path="https://cc-dam.imgix.net/documentation/images/editing/save-changes/2024-03/CloudCannon-Documentation-Save-Changes-Commit-Message.png" alt="The Review changes modal with a commit message describing the changes made to the site." title="Commit messages in the Review changes modal with commit message field" type="ui-snippet"/>
 
-<comp.Notice info_type="info">
-  If your commit template contains no placeholders, its contents will be loaded directly into a textarea, so the user can edit the whole thing before saving.
-</comp.Notice>
+### Placeholders in commit templates
 
-### Filters
+Commit templates can use placeholders. When CloudCannon reads a placeholder in a commit message, it will replace it with the data it references before sending the commit message to your Git repository.
 
-Placeholders support a number of **filters**. Filters are placed after a `|`, and multiple filters can be applied in sequence. For example, `{title|trim|uppercase}` will trim, then uppercase, a placeholder called "title". Read more about all the available placeholders [here](/documentation/articles/configure-your-template-strings/).
+There are two types of placeholders:
 
-### Examples
+* **Data placeholders** — Replaced with data you have defined. These placeholders use `{ }` brackets.
+* **Fixed placeholders** — Replaced with data defined by CloudCannon. These placeholders use `[ ]` brackets.
 
-The default configuration simply uses the `default_commit_message`&nbsp;placeholder. This will be used if you don't have any configured commit templates.
+For commit message templates, data placeholders reference inputs you have defined in your global configuration file under `commit_templates._inputs`. If you do not define a data placeholder used by your commit template, CloudCannon will make it an empty text input by default (i.e., the `{message}` placeholder).
```

Comment:
> CloudCannon will make it an empty text input by default

If the input type isn't defined, CloudCannon will take a best guess at the intended type based no the placeholder name, falling back to a text input. For example `{date}` will render a date input without any extra configuration.